src/parenttext/firebase_tools.py

```diff
@@ -0,0 +1,222 @@
+from pathlib import Path
+import hashlib
+import base64
+import re
+import uuid
+from google.cloud import storage
+from rpft.google import get_credentials
+
+
+class Firebase:
+    gcs_client: storage.Client  # Get the typing right
+
+    def __init__(self, project_id: str) -> None:
+        """
+        Initializes the Firebase tool.
+
+        Args:
+            project_id (str): Your Google Cloud project ID.
+        """
+
+        gcs_scopes = ["https://www.googleapis.com/auth/cloud-platform"]
+        self.project_id = project_id
+        self.creds = get_credentials(scopes=gcs_scopes)
+        self.gcs_client = storage.Client(
+            project=self.project_id, credentials=self.creds
+        )
+
+    def get_latest_folder_version(self, bucket_name, gcs_base_path) -> int:
+        """Get latest folder version, if folder doesn't exist, return -1.
+
+        Parameters
+        ----------
+        bucket_name : _type_
+            _description_
+        gcs_base_path : _type_
+            _description_
+
+        Returns
+        -------
+        int
+            _description_
+        """
+        # --- Find latest matching directory on the remote ---
+        blob_list = list(self.gcs_client.list_blobs(bucket_name, prefix=gcs_base_path))
+
+        search_pattern = rf"^{gcs_base_path.rstrip('/')}(\d*)/.*$"
+        remote_version_number = -1  # initialize at -1
+        # Get v out as a capture group
+        capture_list = [re.findall(search_pattern, blob.name)[0] for blob in blob_list]
+
+        # Iterate through unique captures
```

Comment:
This comment describes what is obvious from the code, so I think it should be removed.

My personal preference is to only comment as a last resort. I would comment if the code is difficult to understand or might be surprising to a reader and it is not possible to refactor the code to make it easier to understand, or more intuitive.

It is also worth commenting to record why the code is the way it is, if it might be controversial.

I have singled out this particular comment, but there are others like it. Would you please review the other comments and remove if appropriate?

Reply:
I've found some significant functionality elsewhere to be non-obvious and not commented. I try to comment so folks who are unaware of non-basic syntax can understand. Particularly cases where I'm doing a one liner. `set` is uncommonly used so I think this method of iterating over unique elements may be confusing.
```suggestion
        # Iterate through unique captures, as the capture list will contain a duplicate for each file in the folder
```